llvm/lib/Transforms/IPO/ArgumentPromotion.cpp

```diff
@@ -139,6 +141,12 @@ doPromotion(Function *F, FunctionAnalysisManager &FAM,
     } else if (I->use_empty()) {
       // Dead argument (which are always marked as promotable)
       ++NumArgumentsDead;
+      ORE.emit([&]() {
+        return OptimizationRemark(DEBUG_TYPE, "ArgumentRemoved", F)
+               << ore::NV("Function", F) << " removing argument "
```

Comment:
Nitpick: function name is printed twice, first for `OptimizationRemark`, next for `NV`, is it really necessary?

Reply:
Good suggestion. The second function name is not needed. Will fix.